packages/slate-locales-plugin/index.js

```diff
@@ -0,0 +1,183 @@
+        getSchema(
+          dirent,
+          path.resolve(compilation.options.context, 'sections'),
+          compilation,
+          dirents,
+        );
+      }
+    });
+  }
+};
+
+async function getSchema(liquidDirent, directoryPath, compilation, dirents) {
+  if (liquidDirent.name === 'template.liquid') {
+    const outputKey = `../sections${
+      directoryPath.split('/sections')[1]
+    }.liquid`;
+
+    if (
+      dirents.find((obj) => obj.name === 'locales' && obj.isDirectory()) &&
+      dirents.find((obj) => obj.name === 'schema.json' && obj.isFile())
+    ) {
+      compilation.assets[outputKey] = new ConcatSource(
+        new RawSource(
+          fs.readFileSync(path.resolve(directoryPath, 'template.liquid')),
+        ),
+        new RawSource(
+          `{% schema %}\n ${JSON.stringify(
+            createMainSchema(
+              combineLocales(path.resolve(directoryPath, 'locales')),
+              path.resolve(directoryPath, 'schema.json'),
+            ),
+            null,
+            2,
+          )}{% endschema %}`,
+        ),
+      );
+    } else if (
+      dirents.find((obj) => obj.name === 'schema.json' && obj.isFile())
+    ) {
+      compilation.assets[outputKey] = new ConcatSource(
+        new RawSource(
+          fs.readFileSync(path.resolve(directoryPath, 'template.liquid')),
+        ),
+        new RawSource(
+          `{% schema %}\n${fs.readFileSync(
+            path.resolve(directoryPath, 'schema.json'),
+          )}{% endschema %}`,
+        ),
+      );
+    } else {
+      compilation.assets[outputKey] = new RawSource(
+        fs.readFileSync(path.resolve(directoryPath, 'template.liquid')),
+      );
+    }
+  } else {
+    const liquidContent = await fs.readFile(
+      path.resolve(directoryPath, liquidDirent.name),
+      'utf8',
+    );
+    const outputKey = `../sections${
+      directoryPath.split('/sections')[1] === ''
+        ? `/${liquidDirent.name}`
+        : `${directoryPath.split('/sections')[1]}/${liquidDirent.name}`
+    }`;
+    if (
+      liquidContent.match(/{%\s*schema\s*%}([\s\S]+){%\s*endschema\s*%}/gim) ||
+      !dirents.find(
+        (obj) => obj.name === liquidDirent.name.replace('.liquid', '.json'),
+      )
+    ) {
+      compilation.assets[outputKey] = new RawSource(liquidContent);
+    } else {
+      compilation.assets[outputKey] = new ConcatSource(
+        new RawSource(liquidContent),
+        new RawSource(
+          `{% schema %}\n${fs.readFileSync(
+            path.resolve(
+              directoryPath,
+              liquidDirent.name.replace('.liquid', '.json'),
+            ),
+          )}{% endschema %}`,
+        ),
+      );
+    }
+  }
+}
+
+async function traverseDirectory(dirent, directoryPath, compilation) {
+  const dirents = await fs.readdir(path.resolve(directoryPath, dirent.name), {
+    withFileTypes: true,
+  });
+  dirents.forEach((subdirent) => {
+    if (subdirent.isDirectory() && subdirent.name !== 'locales') {
```

Comment:
I think we should only allow one layer of nesting, no? What's the point of supporting more?

Reply:
Wasn't initially sure if people nested but if you think we don't need further nesting then that should make it simpler